docs/PARSING_DATA.md

```diff
@@ -0,0 +1,216 @@
+# Parsing Data
+
+Telegraf has the ability to take data in a variety of formats. Telegraf requires
+configuration from the user in order to correctly parse, store, and send the
+original data. Telegraf does not take the raw data and maintain it internally.
+
+Telegraf uses [line protocol][] to store data. This means that data needs to be
+broken up into a metric name, tags, fields, and a timestamp. While all of these
+options are not required, they are available to the user and be necessary to
+define to ensure the data is correctly represented.
+
+[line protocol]: https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/
+
+## Parsers
+
+The first step is to determine which parser to use. Look at the list of
+[parsers][] and find one that will work with the user's data. This is generally
+straightforward as the data-type will only have one parser that is actually
+applicable to the data.
+
+[parsers]: https://github.com/influxdata/telegraf/tree/master/plugins/parsers
+
+### JSON parsers
+
+There is an exception when it comes to JSON data. Instead of a single parser,
+there are three different parsers capable of reading JSON data:
+
+* `json`: This parser is great for flat JSON data. If your JSON is more complex
+  and for example, has other objects or nested arrays, then do not use this and
+  look at the other two options.
+* `json_v2`: The v2 parser was created out of the need to parse JSON objects. It
+  can take on more advanced cases, at the cost of additional configuration.
+* `xpath`: The xpath parser is the most capable of the three options. While the
+  xpath name may imply XML data, it can parse a variety of data types using
+  XPath expressions.
```

Comment:
```suggestion
* `xpath_json`: The xpath parser is the most capable of the three options. While
  the xpath name may imply XML data, it can parse a variety of data types using
  XPath expressions.
```